[Ngoh Wei Yue] iP

nweiyue opened the pull request.

Commit:
Add Gradle support

Commit:
Implement Skeletal Version

Commit:
Add ability to add and list tasks

Commit:
Add ability to mark tasks as done.

Commit:
Add support for tracking three types of tasks

Commit:
Automated Text UI Testing

Commit:
Allow Duke to deal with errors

Commit:
Add support for deleting task

nweiyue committed:
Change method signature for handleTaskCommand

nweiyue committed:
Add Java Enums

nweiyue committed:
Allow Duke to save files in hard disk

nweiyue committed:
Allow Duke to understand Dates and Times

nweiyue committed:
Allow Duke to understand Date and Time

nweiyue committed:
Allow Duke to store tasks in hard disk

nweiyue committed:
Resolve merge conflict

nweiyue committed:
Use more OOP

Change class Data to Storage for better readability
Add classes:
* Ui: to interact with users
* Parser: to make sense of user inputs
* Command and *Command: to extract out commands logic

nweiyue committed:
Organise java classes into packages

nweiyue committed:
Add JUnit tests

nweiyue committed:
Add JavaDoc comments

nweiyue committed:
Tweak code to follow coding standards

nweiyue committed:
Allow users to find tasks using a keyword

nweiyue committed:
Merge branch 'branch-A-JavaDoc' into master

nweiyue committed:
Resolve merge conflicts

nweiyue committed:
Allow users to find tasks using keywords

nweiyue committed:
Allow users to find tasks using keywords/keyphrases

nweiyue committed:
Update text-ui-test file to test new command: find

nweiyue committed:
Merge remote-tracking branch 'origin/add-gradle-support' into master

nweiyue committed:
Set up Gradle

Automate project builds using Gradle and allow Gradle to run unit test.

nweiyue committed:
Merge branch 'branch-A-Gradle' into master

nweiyue committed:
Use CheckStyle

nweiyue committed:
Merge branch 'branch-A-CheckStyle' into master

nweiyue committed:
Storage class: Add new case for switch statement

nweiyue committed:
Implement GUI for Duke

Add new class: CommandResponse to be return whenever a Command is
executed.

nweiyue committed:
Merge branch 'branch-A-Level-10' into master

kswk commented on src/main/java/duke/command/CommandResponse.java:
```diff
@@ -0,0 +1,56 @@
+     * @return responseMessage.
+     */
+    public String getResponseMessage() {
+        return this.responseMessage;
```
Is the "this" keyword necessary?

kswk commented on src/main/java/duke/command/CommandResponse.java:
```diff
@@ -0,0 +1,56 @@
+     * @return shouldExit.
+     */
+    public boolean getShouldExit() {
+        return this.shouldExit;
```
Is the "this" keyword necessary?

kswk commented on src/main/java/duke/command/CommandResponse.java:
```diff
@@ -0,0 +1,56 @@
+            return true;
+        } else if (obj instanceof CommandResponse) {
+            CommandResponse cr = (CommandResponse) obj;
+            return cr.responseMessage.equals(this.responseMessage) && cr.shouldExit == this.shouldExit;
```
Is the "this" keyword necessary?

kswk commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,74 @@
+            return true;
+        } else if (obj instanceof DeleteCommand) {
+            DeleteCommand c = (DeleteCommand) obj;
+            return c.taskNumber == this.taskNumber && c.getIsExit() == this.getIsExit();
```
Is the "this" keyword necessary?

kswk commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,54 @@
+            return true;
+        } else if (obj instanceof DoneCommand) {
+            DoneCommand c = (DoneCommand) obj;
+            return c.taskNumber == this.taskNumber && c.getIsExit() == this.getIsExit();
```
Is the "this" keyword necessary?

kswk commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,69 @@
+            return true;
+        } else if (obj instanceof FindCommand) {
+            FindCommand c = (FindCommand) obj;
+            return c.content.equals(this.content) && c.getIsExit() == this.getIsExit();
```
Is the "this" keyword necessary?

kswk commented on src/main/java/duke/command/RetrieveCommand.java:
```diff
@@ -0,0 +1,94 @@
+            return true;
+        } else if (obj instanceof RetrieveCommand) {
+            RetrieveCommand c = (RetrieveCommand) obj;
+            return c.date.isEqual(this.date) && c.getIsExit() == this.getIsExit();
```
Is the "this" keyword necessary?

kswk commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,69 @@
+     * @return A LocalDateTime object.
+     */
+    public LocalDateTime getDateTime() {
+        return this.at;
```
Is the "this" keyword necessary?

kswk commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,83 @@
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
```
Is the "this" keyword necessary?

kswk commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,83 @@
+     * Marks the Task as done by changing the value of isDOne to true.
+     */
+    public void completeTask() {
+        this.isDone = true;
```
Is the "this" keyword necessary?

kswk commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,83 @@
+     * @return Description of the Task.
+     */
+    public String getDescription() {
+        return this.description;
```
Is the "this" keyword necessary?

kswk commented on src/main/java/duke/ui/MainWindow.java:
```diff
@@ -0,0 +1,70 @@
+package duke.ui;
+
+import duke.Duke;
+import duke.command.CommandResponse;
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
```
I like the lexicographic way in which you arrange your import statements in all your classes, maybe it will look neater if there is a line break in between the the import statements at line 4 and 5? 
Also a swap in lines 3 and 4 to make it look consistent. 😄

nweiyue replied:
Thank you for the suggestions! I will look into them.

kswk reviewed (commented):
LGTM, just some nits to fix.

zhizhizhizhizhi commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -42,6 +52,11 @@ public void save(TaskList list) {
+     *
+     * @return Nothing.
+     */
     public ArrayList<Task> load() {
```
Perhaps you can try to keep this method shorter by abstracting the code for string splitting and comparing to the Parser class.

nweiyue committed:
Update JavaFx API version

zhizhizhizhizhi commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -87,50 +106,50 @@ public static Command parseDone(String input, TaskList tasks) throws DukeExcepti
      * @throws DukeException If user input is not valid.
      */
     public static Command parseAddTask(String input) throws DukeException {
```
This method does similar things to the load() method in Storage. Maybe you can extract the logic into a new method in this class and apply it in both classes so the code can be shortened.

nweiyue replied:
Thank you for the suggestions! I will try to incorporate them in my future commits.

zhizhizhizhizhi reviewed (commented):
Your code is well documented and easy to read. I could not really find any issue with it. Checkstyle shows that you have no styling error too. Good job!

nweiyue committed:
Use Assertions

nweiyue committed:
Improve Code Quality

nweiyue committed:
Merge pull request #1 from nweiyue/branch-A-Assertions

Use Assertions

nweiyue committed:
Resolve merge conflict with branch master

nweiyue committed:
Merge pull request #2 from nweiyue/branch-A-CodeQuality

Improve Code Quality

nweiyue committed:
Update tasks.txt

nweiyue committed:
Fix bug from DeleteCommand

nweiyue committed:
Set up CI

nweiyue committed:
Allow Duke to sort Task

nweiyue committed:
Merge pull request #3 from nweiyue/branch-C-Sort

Allow Duke to sort Task

nweiyue committed:
Remove windows-latest from gradle.yml

nweiyue committed:
Use Lambdas

nweiyue committed:
Merge pull request #4 from nweiyue/branch-A-Lambdas

Use Lambdas

nweiyue committed:
Use Streams

nweiyue committed:
Fix merge conflicts with branch-A-Streams

nweiyue committed:
Merge pull request #5 from nweiyue/branch-A-Streams

Branch-A-Streams

nweiyue committed:
Improve GUI using CSS

GUI is rather bland.

A more beautiful GUI will improve the UX.

Let's,
* Make changes to color, paddings and shape of chat bubble.
* Change profile pictures of Duke and User.
* Add background image.

nweiyue committed:
Add title for GUI

nweiyue committed:
Improve Code Quality

nweiyue committed:
Remove tab infront of String output.

nweiyue committed:
Create TimedTask Class

Deadline and Event extends Task class that does not have a LocalDateTime
attribute.

This causes excessive code duplication in checkIfSameDate() in the
RetrieveCommand class.

Let's,
* create a new abstract class called TimedTask that extends Task and
contain a LocalDateTime attirbute
* extend Deadline and Event class from TimedTask
* update checkIfSameDate() in RetrieveCommand class

Using polymorphism reduces code duplication and allow the program
to be easily expandable.

nweiyue committed:
Merge the logic of adding Task into one

Adding of different type of Task is done seperately, resulting in
code duplication.

Unifying the adding of Task improves the code quality.

Let's,
* create a new AddTaskCommand to replace AddTodoCommand,
AddDeadlineCommand and AddEventCommand
* update the parser method to return only AddTaskCommand
* update test methods

nweiyue committed:
Add Help Command

Display a list of commands that Duke can understand.

nweiyue committed:
Add representative screenshot

nweiyue committed:
Update Ui.png

nweiyue committed:
Change keywords to keyword

nweiyue committed:
Update Junit test for HelpCommand

nweiyue committed:
Update README.md

nweiyue committed:
Update README.md

nweiyue committed:
Update StartPage.png

nweiyue committed:
Set theme jekyll-theme-cayman

nweiyue committed:
Update build.gradle

nweiyue committed:
Allow Duke to create directory

nweiyue committed:
Resolve invalid task number bug

nweiyue committed:
Trim whitespace for commands

nweiyue committed:
Update error messages

ParseAddTask does not take into account if there is
missing description or date/time.

ParseAddTask returns error message asking to input
date/time if either description/dateTime is missing.

Let's,
* Update the error message to a more general one by
asking user to input both description and date/time if
either one is missing by adding InvalidTaskContentException
* Remove NoTaskDateTimeException
* Add NoTaskDate for invalid RetrieveCommand

nweiyue committed:
Resize GUI window

nweiyue committed:
Update README.md

nweiyue committed:
Add TaskType Class

nweiyue committed:
Add Exceptions to TaskList Class

nweiyue committed:
Update JavaDocs

nweiyue committed:
Handle corrupt data file

nweiyue committed:
Resize GUI window

nweiyue committed:
Storage: Handle DateTimeParseException() in load()

nweiyue committed:
Update README.md

nweiyue committed:
Save tasks after sorting

nweiyue committed:
Update README.md

nweiyue committed:
Update README.md
